python/python/lance/dataset.py

```diff
@@ -595,6 +599,12 @@ def delete(self, predicate: Union[str, pa.compute.Expression]):
             predicate = str(predicate)
         self._ds.delete(predicate)
 
+    def count_deleted_rows(self) -> int:
+        """
+        Counts the deleted rows in the fragments from the dataset.
```

Comment:
We may need more explanation here.  This is the count of "soft deleted" rows.  Rows that have been deleted but remain in the dataset.  It will not show entirely deleted fragments.  It also won't show soft deleted rows that were compacted away.

Reply:
Thanks good call -- I updated the method description to reflect the behaviour